control-base/src/main/java/fi/nls/oskari/control/layer/GetAnalysisLayersHandler.java

```diff
@@ -52,18 +46,11 @@ public void handleAction(ActionParameters params) throws ActionException {
 
         final User user = params.getUser();
         final List<Analysis> list = analysisService.getAnalysisByUid(user.getUuid());
-        Set<String> publishPermission = permissionsService.getResourcesWithGrantedPermissions(ResourceType.analysislayer, user, PermissionType.PUBLISH);
-        Set<String> downloadPermission = permissionsService.getResourcesWithGrantedPermissions(ResourceType.analysislayer, user, PermissionType.DOWNLOAD);
-        JSONObject baseOptions = AnalysisDataService.getBaseLayer().getOptions();
         for(Analysis a: list) {
             // Parse analyse layer json out analysis
-            final JSONObject analysisLayer = AnalysisHelper.getlayerJSON(a, baseOptions);
-            final String permissionKey = "analysis+" + a.getId();
-
-            JSONHelper.putValue(analysisLayer, "permissions",
-                    AnalysisHelper.getAnalysisPermissions(
-                            publishPermission.contains(permissionKey),
-                            downloadPermission.contains(permissionKey)));
+            final JSONObject analysisLayer = AnalysisDataService.parseAnalysis2JSON(a, null);
+            // user's own analysis, add default permissions
+            LayerJSONFormatterANALYSIS.setDefaultPermissions(analysisLayer);
```

Comment:
The permissions for analysis layers are inherited from the source layer so we can't just allow all. Otherwise user could do a buffer with 0 to copy features from a layer that can't be published to an analysislayer that could be published.